### PR TITLE
feat: PRCE-compliant RegEx updates for GitHub Linguist

### PR DESCRIPTION
**Pull request by randi274**
### What does this PR do?
This PR updates certain RegEx elements to be PCRE RegEx compliant (specifically: changing `-` to `_` in named capture group identifiers).

The change is in [this commit](https://github.com/forcedotcom/apex-tmLanguage/commit/2e43ac7ca55edfc734cf0832ae12e4ac3005b675).

This is needed so that we can get the tmLanguage definition into [GitHub's Linguist for language detection](https://github.com/github/linguist/pull/6198) and syntax highlighting.

The other changes outside of the above linked commit are either dependency updates or reformatting of files (b/c of changed Prettier behavior).

### What issues does this PR fix or reference?
#54

### CODE REVIEW

**Comment by muenzpraeger** [comment]
@randi274 Any updates on this? TY

File: grammars/apex.tmLanguage [hunk 1]
```diff
@@ -1914,15 +1914,15 @@
       <dict>
         <key>begin</key>
         <string>(?x)
-(?&lt;type-name&gt;
+(?&lt;type_name&gt;
```
Comment (jeffb-sfdc) [inline grammars/apex.tmLanguage:1917]: Dumb question: should everything between a `<string></string>` be changed from `foo-bar` to `foo_bar`?  If yes, what about line 1909, line 1899, etc...?

File: grammars/apex.tmLanguage [hunk 2]
```diff
@@ -1914,15 +1914,15 @@
...
   (?:
     (?:
       (?:(?&lt;identifier&gt;@?[_[:alpha:]][_[:alnum:]]*)\s*\:\:\s*)? # alias-qualification
-      (?&lt;name-and-type-args&gt; # identifier + type arguments (if any)
+      (?&lt;name_and_type_args&gt; # identifier + type arguments (if any)
```
Comment (jeffb-sfdc) [inline grammars/apex.tmLanguage:1921]: Another dumb questions: Should `alias-qualification` on line 1920 be `alias_qualification`?

File: grammars/apex.tmLanguage (hunk 1 above)
Comment (randi274) [reply grammars/apex.tmLanguage:1917]: @muenzpraeger can you help answer this, given the update needs for Github Linguist?

File: grammars/apex.tmLanguage (hunk 2 above)
Comment (randi274) [reply grammars/apex.tmLanguage:1921]: @muenzpraeger same question here

File: grammars/apex.tmLanguage (hunk 1 above)
Comment (muenzpraeger) [reply grammars/apex.tmLanguage:1917]: There are no dumb questions. ;-) Especially not with this change. It took me a while to find [this small statement in the PCRE docs for named capture groups](https://www.pcre.org/current/doc/html/pcre2pattern.html#SEC16).

> When PCRE2_UTF is not set, they may contain only ASCII alphanumeric characters and underscores, but must start with a non-digit.

So yes, the change only applies to `(?<the_name>)`.

Also the source of the changes is [src/apex.tmLanguage.yml](https://github.com/forcedotcom/apex-tmLanguage/pull/57/files#diff-733b6ef5f9983ffaefca26f17d76ae4264d95125db0d311bea2c34202e68836f), which is the source definition for this TextMate grammar. Looking at that will make it clearer.

The grammar outputs in the grammars folder are [auto-generated via gulp](https://github.com/forcedotcom/apex-tmLanguage/blob/main/gulpfile.js#L34-L41).

**Review by jeffb-sfdc — APPROVED** [review]

**Comment by dschach** [comment]
@randi274 Thank you for doing this. It will make my port to a new version of highlightjs-apex much easier.

There are some extraneous bits in the regex (or there were - I haven't checked lately) that we can't do in Apex; as I get reacquainted with this codebase I'll raise a PR or an issue to remove those... which may be a bad idea, but I figure why not see what you and the hive mind think?

Thanks again. It's great to see how Apex is defined as a syntax.

**Pull request merged**